PHP7.4 as minimum required version

DanielBadura opened the pull request:
PHP7.2 is running EOL in 1 month and with this in mind I updated the minimum required PHP version. I went up to `7.4.*` as I think its not really worth to also maintain for PHP 7.3. But we can, if you want, go down to 7.3.

I also updated all dependencies to the newest minor versions and PHPUnit up to `^9.4.0`.

DanielBadura commented on composer.json:
```diff
@@ -26,19 +26,19 @@
     ],
     "require": {
-        "php": "^7.2",
+        "php": "7.4.*",
```
not sure if we should already open up for php8

Ocramius replied:
Not unless we explicitly test for it

DanielBadura commented:
I dont really understand why scrutinizer is failing. Maybe someone can give me a hint? 🦑

Ocramius commented:
We need to upgrade phpcs, as it seems:

```
PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /home/scrutinizer/.analysis/phpcs/vendor/squizlabs/php_codesniffer/src/Config.php on line 399
````

I don't see it being installed via composer though, but I'm OK with scrutinizer-ci failing, as it really should be slowly phased out.

Ocramius commented on .travis.yml:
```diff
@@ -6,12 +6,10 @@ git:
-            - COMPOSER_FLAGS="--prefer-stable --prefer-lowest"
-            - COVERAGE=1
-        - php: 7.3
         - php: 7.4
```
There should be a `7.4` entry without `--prefer-stable --prefer-lowest`

Ocramius replied:
This can now be completely scrapped, since we have GitHub actions covering all cases

DanielBadura replied:
Only case missing would be the nightly build for php8 as mentioned in #208 but this should not be considered here in this patch. I will rebase this today 👍

DanielBadura commented:
> We need to upgrade phpcs, as it seems:
> 
> ```
> PHP Deprecated:  Array and string offset access syntax with curly braces is deprecated in /home/scrutinizer/.analysis/phpcs/vendor/squizlabs/php_codesniffer/src/Config.php on line 399
> ```
> 
> I don't see it being installed via composer though, but I'm OK with scrutinizer-ci failing, as it really should be slowly phased out.

Yes that was confusing me. It errors because of phpcs but i did not see it as a dep. I will add phpcs soon in another PR 😄

DanielBadura commented:
@Ocramius this should be ready now. I rebased to get the workflows in and removed travis and appveyor - forgot it in the last patch. Also removed php7.2 and 7.3 from the ci

DanielBadura commented:
Noticing that appveyor still runs on this PR but now failing. Did i miss something?

Ocramius commented:
> Noticing that appveyor still runs on this PR but now failing. Did i miss something?

I would really just turn off all the other CI tools, now that we have github actions (which can test things on windows too)

DanielBadura commented:
Yeah i wanted to and removed the config but its still running

Ocramius commented:
@DanielBadura feel free to drop it: disabling it is up to @maglnet though, as I don't have access to repository settings.

DanielBadura commented:
@Ocramius appveyor config is already dropped. I think this  patch is ready as it is.

Ocramius commented:
:ship:  then :)
